src/rebuild/scan.c

```diff
@@ -516,8 +516,13 @@ obj_reclaim(struct pl_map *map, uint32_t layout_ver, uint32_t new_layout_ver,
 	if (rc != 0)
 		return rc;
 
-	still_needed = pl_obj_layout_contains(rpt->rt_pool->sp_map, layout, myrank, mytarget,
-					      oid.id_shard);
+	/* If there are further targets failure during reintegration/extend/drain,
+	 * rebuild will choose replacement targets for the impacted objects anyway,
+	 * so we do not need reclaim these impacted shards by @ignore_rebuild_shard.
+	 */
+	still_needed = pl_obj_layout_contains(rpt->rt_pool->sp_map, layout, myrank,
+					      mytarget, oid.id_shard,
+					      rpt->rt_rebuild_op == RB_OP_RECLAIM ? true : false);
```

Comment:
RB_OP_FAIL_RECLAIM  is used to clear all objects "half" rebuilt, and RB_OP_RECLAIM is used to clear the stale object after reintegration/drain.

In this case RB_OP_RECLAIM should not ignore the rebuilding shard, since "DOWN" object has been resolved during reintegration. 

But for RB_OP_FAIL_RECLAIM, it should clear all objects which are half rebuilt, i.e. all rebuilding/reintegrating shards be deleted.